File: Difficulty: Medium/K closest elements/k-closest-elements.py
import heapq

class Solution:
    def printKClosest(self, arr, k, x):
        heap = []

        for num in arr:
            if num == x:
                continue
            diff = abs(num - x)
            heapq.heappush(heap, (diff, -num))

        ans = []
        for _ in range(k):
            if heap:
                ans.append(-heapq.heappop(heap)[1])

        return ans

File: Difficulty: Medium/K closest elements/k-closest-elements.cpp
class Solution {
  public:
    vector<int> printKClosest(vector<int> arr, int k, int x) {
        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> pq;

        for (int num : arr) {
            if (num == x) continue;
            int diff = abs(num - x);
            pq.push({diff, -num});  
        }

        vector<int> ans;

        while (k-- && !pq.empty()) {
            ans.push_back(-pq.top().second);  
            pq.pop();
        }

        return ans;
    }
};
